articles/password-checker-javascript/index.md

```diff
@@ -0,0 +1,230 @@
+|[A-Za-z0-9]|Checks all alphanumeric characters|
+|[a-z]|Checks for lowercase letters|
+|[A-Z]|Checks for uppercase letters|
+|x(?=y)|Returns x if and only if it is followed by y|
+|**.**|Checks for any single character except line terminators |
+|x*|Checks for x 0 or more times|
+
+
+#### The JavaScript program
+
+We are going to discuss how to check the strength of a password that a user enters based on:
+
+- Is at least 8 characters long(`?=.{8,}`)
+
+- Has at least one uppercase letter(`?=.*[A-Z]`)
+
+- Has at least one lowercase letter(`?=.*[a-z]`)
+
+- Has at least one digit(`?=.*[0-9]`)
+
+- Has at least one special character(`[^A-Za-z0-9]`)
+
+
+We are going to have three levels:
+
+- **Strong**: The password has to meet all the requirements to be strong.
+
+```java
+(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[^A-Za-z0-9])(?=.{8,})
+```
+
+![Strong password](/engineering-education/password-checker-javascript/strong-password.png)
+
+- **Medium**: The password can only be medium either if it is at least 6 characters long and meets all the other requirements or has no digit but meets the rest of the requirements.
+
+```java
+((?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[^A-Za-z0-9])(?=.{6,}))|((?=.*[a-z])(?=.*[A-Z])(?=.*[^A-Za-z0-9])(?=.{8,}))
+```
+
+![Medium password](/engineering-education/password-checker-javascript/medium-password.png)
+
+- **Weak**: If the entered password does not meet the strong or medium-level requirements then it is deemed weak.
+
+![Weak password](/engineering-education/password-checker-javascript/weak-password.png)
+
+
+We will use Bootstrap 5 for UI styling.
+
+A badge is used to show the password strength and color. Green for strong, Blue for medium and Red for weak.
+
+#### HTML Code
+
+```html
+<html>
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <!-- Bootstrap 5 link -->
+    <link href= "https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-giJF6kkoqNQ00vy+HMDP7azOuL0xtbfIcaT9wjKHr8RbDVddVHyTfAAsrekwKmP1" crossorigin="anonymous">
+
+<!-- CSS Styling -->
+<style>
+    .passwordInput{
+        margin-top: 5%; 
+        text-align :center;
+    }
+
+    .displayBadge{
+        margin-top: 5%; 
+        display: none; 
+        text-align :center;
+    }
+</style>
+    
+<body>
+    <p>
+        <div class="d-flex text-danger justify-content-center">
+            <ul style="list-style-type:none">
+                <li class="fs-4" style="margin-top: 10%; text-align :center">
+                    Password Strength Checker
+                </li>
+                <li>
+                    <input id="0101" class="form-control passwordInput" type="password" placeholder="Type your password" id="inputPassword">
+                </li>
+                <li>
+                    <span id="0102" class="badge displayBadge">Weak</span>
+                </li>
+            </ul>
+        </div>  
+    </p>     
+</body>
+</html>
+```
```

Comment:
Can you briefly explain what the HTML code does? The elements you are adding and why you are adding them.